worker/uniter/operation/runhook.go

```diff
@@ -129,14 +129,20 @@ func (rh *runHook) Execute(state State) (*State, error) {
 	}.apply(state), err
 }
 
-func (rh *runHook) beforeHook() error {
+func (rh *runHook) beforeHook(state State) error {
 	var err error
 	switch rh.info.Kind {
 	case hooks.Install:
-		err = rh.runner.Context().SetUnitStatus(jujuc.StatusInfo{
-			Status: string(status.Maintenance),
-			Info:   status.MessageInstallingCharm,
-		})
+		// If the charm has already updated the unit status in a previous hook,
+		// then don't overwrite that here - just set the executing status message.
+		if state.StatusSet {
+			err = rh.callbacks.SetExecutingStatus(status.MessageInstallingCharm)
```

Comment:
I think this is pointless, because it's going to be replaced immediately by the SetExecutingStatus call in Execute.